src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -207,6 +207,55 @@ void regclass_Core(py::module m) {
             :rtype: openvino.runtime.CompiledModel
         )");
 
+    cls.def(
+        "compile_model",
+        [](ov::Core& self,
+           const py::object& model_buffer,
+	   const py::object& weight_buffer,
+           const std::string& device_name,
+           const std::map<std::string, py::object>& properties) {
+            std::stringstream _stream;
+            _stream << model_buffer.cast<std::string>();
+
+            py::buffer_info info;
+	    if (!py::isinstance<py::none>(weight_buffer)) {
+                auto p = weight_buffer.cast<py::bytes>();
+                info = py::buffer(p).request();
+            }
+            size_t bin_size = static_cast<size_t>(info.size);
+            ov::Tensor tensor(ov::element::Type_t::u8, {bin_size});
+            // if weights are not empty
+            if (bin_size) {
+                const uint8_t* bin = reinterpret_cast<const uint8_t*>(info.ptr);
+                std::memcpy(tensor.data(), bin, bin_size);
```

Comment:
Can we create tensor with input buffer rather than memcpy it to tensor?

Reply:
how can we guarantee lifetime of  `const py::object& weight_buffer` ? It can be destroyed on Python side by GC

We need a way to hold it on C++ side.

@jiwaszki any advices here?

Reply:
Thanks for your feedback. Since I am not expert for python binding, will be very appreciate if any suggestions to better handle it! @jiwaszki.

Reply:
@jiwaszki, thanks for your great support for better handle weights lifetime and avoid unnecessary copy for better perf.   
@riverlijunjie, @ilya-lavrenov, I update Jan's suggestions in my PR as follow: 
- Weights lifetime handling: store weights in Python wrapper, so that the weights are tied to lifetime of CompiledModel itself.
- Assigning weights to other variable does not create a copy. Bytes are immutable by design which makes it easy to share them, plus each assignment result in overriding of value.